hnn_core/gui.py

```diff
@@ -0,0 +1,1028 @@
+
+    cell_types = ['L5_pyramidal', 'L2_pyramidal', 'L5_basket', 'L2_basket']
+    rate_constant = dict()
+    for cell_type in cell_types:
+        rate_constant[f'{cell_type}'] = FloatText(
+            value=default_data['rate_constant'][cell_type],
+            description=f'{cell_type}:',
+            layout=layout,
+            style=style)
+
+    widgets_list, widgets_dict = _get_cell_specific_widgets(
+        layout,
+        style,
+        location,
+        data={
+            'weights_ampa': default_weights_ampa,
+            'weights_nmda': default_weights_nmda,
+            'delays': default_delays,
+        },
+    )
+    widgets_dict.update({'rate_constant': rate_constant})
+    widgets_list.extend([HTML(value="<b>Rate constants</b>")] +
+                        list(widgets_dict['rate_constant'].values()))
+
+    drive_box = VBox([tstart, tstop, seedcore] + widgets_list)
+    drive = dict(
+        type='Poisson',
+        name=name,
+        tstart=tstart,
+        tstop=tstop,
+        rate_constant=rate_constant,
+        seedcore=seedcore,
+        location=location,  # notice this is a widget but a str!
+    )
+    drive.update(widgets_dict)
+    return drive, drive_box
+
+
+def _get_evoked_widget(
+    name,
+    layout,
+    style,
+    location,
+    data=None,
+    default_weights_ampa=None,
+    default_weights_nmda=None,
+    default_delays=None,
+):
+    default_data = {
+        'mu': 0,
+        'sigma': 1,
+        'numspikes': 1,
+        'seedcore': 14,
+    }
+    if isinstance(data, dict):
+        default_data.update(data)
+    kwargs = dict(layout=layout, style=style)
+    mu = FloatText(value=default_data['mu'],
+                   description='Mean time:',
+                   **kwargs)
+    sigma = FloatText(value=default_data['sigma'],
+                      description='Std dev time:',
+                      **kwargs)
+    numspikes = IntText(value=default_data['numspikes'],
+                        description='No. Spikes:',
+                        **kwargs)
+    seedcore = IntText(value=default_data['seedcore'],
+                       description='Seed: ',
+                       **kwargs)
+
+    widgets_list, widgets_dict = _get_cell_specific_widgets(
+        layout,
+        style,
+        location,
+        data={
+            'weights_ampa': default_weights_ampa,
+            'weights_nmda': default_weights_nmda,
+            'delays': default_delays,
+        },
+    )
+
+    drive_box = VBox([mu, sigma, numspikes, seedcore] + widgets_list)
+    drive = dict(type='Evoked',
+                 name=name,
+                 mu=mu,
+                 sigma=sigma,
+                 numspikes=numspikes,
+                 seedcore=seedcore,
+                 location=location,
+                 sync_within_trial=False)
+    drive.update(widgets_dict)
+    return drive, drive_box
+
+
+def add_drive_widget(
+    drive_type,
+    drive_boxes,
+    drive_widgets,
+    drives_out,
```

Comment:
same comment. Don't break the arguments in different lines for the function definition. I find it hard to read.

Reply:
Actually I think this way makes it more clear to me. But ok I'll follow the existing convention in the repo.

Reply:
Black is probably the culprit. We could discuss alternate code styles but it has to be repo-wide. I just use a `flake8` linter with VS Code that warns me if I'm not conforming to the style ... the "Python" extension includes it, but you may have to enable linting.